### ISSUE TITLE
How can I create a new public function inside SketchView Class

### ISSUE THREAD

**Issue by longnh2604** [daihase/Sketch#12]
Sorry, could you please tell me the reason why I put new public function inside SketchView class file, the SketchView will not load image (image will be white blank). Did I miss something? Thanks so much.
(in my sketchView)
public func checkPublic() {
        
}

(in my viewcontroller)
func setupCanvas() {
        let url = URL(string: media.url)
        let imageV = UIImageView()
        imageV.sd_setImage(with: url, placeholderImage: nil, options: SDWebImageOptions.continueInBackground) { (image, error, cache, url) in
            self.sketchView.loadImage(image: imageV.image!)
            self.sketchView.drawTool = .pen
            self.sketchView.lineWidth = 1.0
        }
    }

**Comment by daihase** [comment]
@longnh2604 
Certainly, when I tried to actually paste the image loaded from the URL using SDWebImage to SketchView, it didn't work.

```
import UIKit
import Sketch
import SDWebImage

class ViewController: UIViewController {
    @IBOutlet weak var sketchView: SketchView!
    private let imageView = UIImageView(frame: CGRect(x: 0, y: 0, width: 200, height: 200))
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        guard let imagUrl = NSURL(string: "https://tech.librastudio.co.jp/wp-content/uploads/2016/02/kinoko_1024.png") as URL? else { fatalError("URL not found") }
        
        imageView.sd_setImage(with: imagUrl, completed: {[weak self](image, err, cacheType, url) in
            self?.sketchView.drawTool = .pen
            self?.sketchView.loadImage(image: self?.imageView.image ?? UIImage())
        })
    }
}
```

Certainly, when I tried to actually paste the image loaded from the URL using SDWebImage to SketchView, it didn't work. I'm copying the UIImage inside Sketch, but it may be necessary to be Deep copy. (Camera images etc. work fine, but ...)

I'd like to continue to investigate more. 🙇 

By the way, as an alternative, you can paint from the top of the photo by placing one UIImageView under SketchView.

**Comment by daihase** [comment]
@longnh2604
How embarrassing!
It seemed that my code displayed the image from the Internet was too small, and it was displayed without problems.

So please try the above code.

**Comment by longnh2604** [comment]
Thanks for your hard work. Right now with my above code, I can load the image from the http link with no problem.
Just only one thing I don't know why is when I have just added the new public function inside your sketchView file, the image from SketchView will not load properly.
public func checkPublic() { } without any code inside.

**Comment by longnh2604** [comment]
Thanks so much. You saved my day. I added in the middle of your code and it did not work. But if I've added to the last line of your code as you did, everything is fine.

**Comment by bj1024** [comment]
The same thing happened to me.
Try 'Clean build folder' of xcode projects.
Then it worked fine.

**Comment by daihase** [comment]
@bj1024 
Oh... I didn't realize that.
Thanks 👍 

@longnh2604 
How is this?

**Comment by longnh2604** [comment]
Dear @daihase ,
Sorry, but I got exc-bad-access when call a new one from my ViewController. Do you have any suggest?

**Comment by daihase** [comment]
@longnh2604 
Sorry, I don't  understand because detailed information is nothing.
You need to provide specific information such as which part of the sketch is causing the crash.

And I think it's already different content from this issue, so It's better to make a new issue.
Thank you.
